upup/pkg/fi/cloudup/loader.go

```diff
@@ -214,6 +217,40 @@ func (l *Loader) addAssetCopyTasks(assets []*assets.ContainerAsset, lifecycle *f
 
 		}
 	}
+
+	return nil
+}
+
+// addAssetFileCopyTasks creates the new tasks for copying files.
+func (l *Loader) addAssetFileCopyTasks(assets []*assets.FileAsset, lifecycle *fi.Lifecycle) error {
+	for _, asset := range assets {
+
+		// test if the asset needs to be copied
+		if asset.CanonicalFileURL != nil && asset.FileURL != asset.CanonicalFileURL {
```

Comment:
`asset.FileURL != asset.CanonicalFileURL` is pointer comparison, which is going to bite us as some stage.

Reply:
Probably `continue` if either is nil, then compare fileUrl.String() to canonicalFileUrl.String()

Reply:
I errored if it is a nil.